lib/src/test/kotlin/integration/account/AccountTest.kt

```diff
@@ -0,0 +1,261 @@
+import integration.utils.IntegrationConfig
+import kotlinx.serialization.json.*
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Assumptions.*
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.parallel.Execution
+import org.junit.jupiter.api.parallel.ExecutionMode
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.MethodSource
+import java.nio.file.Path
+import kotlin.io.path.readText
+
+@Execution(ExecutionMode.SAME_THREAD)
+class AccountTest {
+    companion object {
+        @JvmStatic
+        private val config = IntegrationConfig.config
+        private val rpcUrl = config.rpcUrl
+        private val gatewayUrl = config.gatewayUrl
+        private val feederGatewayUrl = config.feederGatewayUrl
+        private val accountAddress = config.accountAddress
+        private val privateKey = config.privateKey
+
+        private lateinit var signer: Signer
+
+        private lateinit var gatewayProvider: GatewayProvider
+        private lateinit var rpcProvider: JsonRpcProvider
+
+        @JvmStatic
+        @BeforeAll
+        fun before() {
+            signer = StarkCurveSigner(
+                privateKey = privateKey,
+            )
+            gatewayProvider = GatewayProvider(
+                feederGatewayUrl,
+                gatewayUrl,
+                StarknetChainId.TESTNET,
+            )
+            rpcProvider = JsonRpcProvider(
+                rpcUrl,
+                StarknetChainId.TESTNET,
+            )
+        }
+        data class AccountAndProvider(val account: Account, val provider: Provider)
+
+        @JvmStatic
+        private fun getProviders(): List<Provider> = listOf(
+            gatewayProvider,
+            rpcProvider,
+        )
+
+        @JvmStatic
+        fun getAccounts(): List<AccountAndProvider> {
+            return listOf(
+                AccountAndProvider(
+                    StandardAccount(
+                        accountAddress,
+                        signer,
+                        rpcProvider,
+                    ),
+                    rpcProvider,
+                ),
+                AccountAndProvider(
+                    StandardAccount(
+                        accountAddress,
+                        signer,
+                        gatewayProvider,
+                    ),
+                    gatewayProvider,
+                ),
+            )
+        }
+
+        @JvmStatic
+        @AfterAll
+        fun after() {}
+    }
+
+    @ParameterizedTest
+    @MethodSource("getAccounts")
+    fun `estimate fee for declare v1 transaction`(accountAndProvider: AccountAndProvider) {
+        assumeTrue(IntegrationConfig.isTestEnabled(requiresGas = false))
+
+        val (account, provider) = accountAndProvider
+        // TODO: find a better account that has a non-changing nonce
```

Comment:
Is it possible to have an account with constant nonce? And if so should we even consider using such an account in tests?

Reply:
Yeah, it could be done via creating account and not using it, and only to use it for this particular kind of test. This test can fail (invalid nonce or invalid signature) because gateway has access to pending transactions and therefore transactions is sent with new nonce, while the old one was expect. 

Since gateway is to be removed anyway, I think this can temporary left as is until then and test this on gateway locally only if we need to.

Reply:
Seems like after enabling access to pending block, our node suffers from the same nonce issue as of now, in other test cases as well 😢